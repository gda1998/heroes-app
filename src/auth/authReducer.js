import { types } from '../types/types';

// const state = {
//     name: 'Gabriel',
//     logged: true
// };

/**
 * Metodo reducer para login y logout de la app
 * 
 * @param  {JSON} state={}
 * @param  {JSON} action
 * @returns  {JSON}
 */
export const authReducer = ( state = {}, action ) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            };
        case types.logout:
            return {
                logged: false
            };
        default:
            return state;
    }
}
