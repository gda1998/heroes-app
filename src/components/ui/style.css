/*Estilos de la opcion Cerrar Sesion*/
.link-logout{
    text-decoration: none;
    color: #9A9BA0;
}
.link-logout:hover{
    text-decoration: none;
    color: #cbccce;
}
