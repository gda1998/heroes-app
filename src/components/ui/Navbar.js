import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

import './style.css';

export const Navbar = () => {

    const { user:{name}, dispatch } = useContext(AuthContext);

    const history = useHistory();

    const handleLogout = () => {
        dispatch({
            type: types.logout
        });
        history.replace('/login');
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">

                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarTogglerDemo01" aria-controls="navbarTogglerDemo01" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                
                <div className="collapse navbar-collapse" id="navbarTogglerDemo01">
                    <Link className="navbar-brand" to="/">
                        Asociaciones
                    </Link>
                    <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                        <li className="nav-item">
                            <NavLink 
                                activeClassName="active"
                                className="nav-link" 
                                exact
                                to="/marvel"
                            >
                                Marvel
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink 
                                activeClassName="active"
                                className="nav-link" 
                                exact
                                to="/dc"
                            >
                                DC
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink 
                                activeClassName="active"
                                className="nav-link" 
                                exact
                                to="/search"
                            >
                                Search
                            </NavLink>
                        </li>
                    </ul>
                    <div className="d-flex">
                        <span className="text-name mx-3 my-2 nav-item">{ name }</span>
                        <button
                            className="btn link-logout"
                            onClick={ handleLogout }
                        >
                            Logout
                        </button>
                    </div>{/* /.d-flex */}

                </div>{/* /.collapse .navbar-collapse */}

            </div>{/*/.container-fluid */}

        </nav>
    );
}