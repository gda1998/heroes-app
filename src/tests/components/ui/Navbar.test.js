import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';

describe('Pruebas en <Navbar />', () => {
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Gabriel',
            logged: true
        }
    };

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>

            <MemoryRouter>

                <Router history={ historyMock }>
                    <Navbar />
                </Router>

            </MemoryRouter>

        </AuthContext.Provider>
    );

    beforeEach( () => jest.clearAllMocks() );

    test('El componente debe de mostrarse correctamente', () => {
        const userName = wrapper.find('.text-name').text().trim();
        expect(wrapper).toMatchSnapshot();
        expect(userName).toBe('Gabriel');
    });

    test('Se debe de hacer un logout correctamente', () => {
        wrapper.find('button').at(1).simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });
    
    
});