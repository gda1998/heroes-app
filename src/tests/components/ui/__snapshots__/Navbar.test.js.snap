// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> El componente debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerDemo01"
    >
      <a
        className="navbar-brand"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Marvel
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
      </ul>
      <div
        className="d-flex"
      >
        <span
          className="text-name mx-3 my-2 nav-item"
        >
          Gabriel
        </span>
        <button
          className="btn link-logout"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</nav>
`;
